server/src/main/java/com/thoughtworks/go/server/newsecurity/filterchains/AuthorizeFilterChain.java

```diff
@@ -80,6 +81,8 @@ public AuthorizeFilterChain(AllowAllAccessFilter allowAllAccessFilter,
 
                 // all apis
                 .addAuthorityFilterChain("/cctray.xml", apiAccessDeniedHandler, ROLE_USER)
+                .addAuthorityFilterChain("/server/messages.json", apiAccessDeniedHandler, ROLE_USER)
+                .addAuthorityFilterChain("/pipelines.json", apiAccessDeniedHandler, ROLE_USER)
```

Comment:
These are "API" it's a internal curl request from the browser, I suppose these should be serviced by the `/**` filter at the bottom.

Reply:
ignore me — i see why this is needed — but we may need this for other ajax URLs from the stage detail and job detail pages too.